### ISSUE TITLE
Short ternary syntax doesn't trigger WordPress.Security.EscapeOutput.OutputNotEscaped sniff

### ISSUE THREAD

**Issue by dingo-d** [WordPress/WordPress-Coding-Standards#1617]
## Bug Description

I was writing some code and noticed I have `echo` with variable (unescaped) that didn't trigger the `WordPress.Security.EscapeOutput.OutputNotEscaped` sniff.

## Minimal Code Snippet
I tested with 

```php
<?php

echo $bla;

$var = 'value';

echo $var ?: '';

$empty_var = '';

echo $empty_var ?: '';
```
And command: 

```sh
cat /Desktop/test.php | /vagrant-local/www/my-project/public_html/wp-content/plugins/my-plugin/vendor/bin/phpcs --stdin-path=/Desktop/test.php -
```

## Error Code
```sh
FILE: /Desktop/test.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 1 | ERROR | Missing file doc comment
   |       | (Squiz.Commenting.FileComment.Missing)
 2 | ERROR | All output should be run through an escaping function
   |       | (see the Security sections in the WordPress Developer
   |       | Handbooks), found '$bla'.
   |       | (WordPress.Security.EscapeOutput.OutputNotEscaped)
----------------------------------------------------------------------
```

Only the first echo is caught.

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.2.10
| PHP_CodeSniffer version | 3.4.0
| WPCS version            | 1.2.0
| WPCS install type       | Composer in a project

**Comment by jrfnl** [comment]
@dingo-d I haven't run any tests, but I suspect it's not so much about the first/subsequent echo statement, as it is about the sniff not having support for short ternary syntax.

The sniff sees, the ternary and only checks if everything in the yes/no conditions is escaped and the `''`in the _no_ condition doesn't need escaping.

Best practice-wise I'm not sure this is the best way to write that code anyway as if you'd want to escape that, you'd need to either escape the complete ternary `esc_htm( $var ?: '')` or escape the condition which would change the meaning of the condition and may result in a redundant function call.

All the same, the `EscapeOutput` sniff should recognize the syntax and should throw an error, so I'm tagging this as a bug.

**Comment by dingo-d** [comment]
> The sniff sees, the ternary and only checks if everything in the yes/no conditions is escaped and the ''in the no condition doesn't need escaping.

This is true, it's triggered when I put some variable instead of `''`, but the issue still remains that if the ternary is truthy the first variable will be echoed out and it should be escaped, so I think this should be triggered as well.

> Best practice-wise I'm not sure this is the best way to write that code anyway

You're right. I could rewrite it to a separate variable, and then just echo and escape that variable, this would be better, didn't think of it.
